src/qibolab/_core/native.py

```diff
@@ -1,92 +1,46 @@
+        return deepcopy(self)
 
     def __call__(self, *args, **kwargs) -> PulseSequence:
-        """Create a sequence for single-qubit rotation.
+        """Create the sequence associated to the gate.
 
         Alias to :meth:`create_sequence`.
         """
         return self.create_sequence(*args, **kwargs)
 
 
-class RxyFactory(Native):
-    """Factory for pulse sequences that generate single-qubit rotations around
-    an axis in xy plane.
-
-    It is assumed that the underlying sequence contains only a single pulse.
-    It is assumed that the base sequence corresponds to a calibrated pi rotation around X axis.
-    Other rotation angles are achieved by scaling the amplitude, assuming a linear transfer function.
-
-    Args:
-        sequence: The base sequence for the factory.
-    """
-
-    def __init__(self, iterable):
-        super().__init__(iterable)
-        cls = type(self)
-        if len(self.channels) != 1:
-            raise ValueError(
-                f"Incompatible number of channels: {len(self.channels)}. "
-                f"{cls} expects a sequence on exactly one channel."
-            )
-
-        if len(self) != 1:
-            raise ValueError(
-                f"Incompatible number of pulses: {len(self)}. "
-                f"{cls} expects a sequence with exactly one pulse."
-            )
-
-        pulse = self[0][1]
-        assert isinstance(pulse, Pulse)
-        expected_envelopes = (Gaussian, Drag)
-        if not isinstance(pulse.envelope, expected_envelopes):
-            raise ValueError(
-                f"Incompatible pulse envelope: {pulse.envelope.__class__}. "
-                f"{cls} expects {expected_envelopes} envelope."
-            )
-
-    def create_sequence(self, theta: float = np.pi, phi: float = 0.0) -> PulseSequence:
-        """Create a sequence for single-qubit rotation.
-
-        Args:
-            theta: the angle of rotation.
-            phi: the angle that rotation axis forms with x axis.
-        """
-        theta, phi = _normalize_angles(theta, phi)
-        ch, pulse = self[0]
-        assert isinstance(pulse, Pulse)
-        new_amplitude = pulse.amplitude * theta / np.pi
-        return PulseSequence(
-            [(ch, replace(pulse, amplitude=new_amplitude, relative_phase=phi))]
-        )
+def _normalize_angles(theta, phi):
+    """Normalize theta to (-pi, pi], and phi to [0, 2*pi)."""
+    theta = theta % (2 * np.pi)
+    theta = theta - 2 * np.pi * (theta > np.pi)
+    phi = phi % (2 * np.pi)
+    return theta, phi
 
 
-class FixedSequenceFactory(Native):
-    """Simple factory for a fixed arbitrary sequence."""
+def rxy(seq: PulseSequence, theta: float = np.pi, phi: float = 0.0) -> PulseSequence:
+    """Create a sequence for single-qubit rotation.
 
-    def create_sequence(self) -> PulseSequence:
-        return deepcopy(self)
+    ``theta`` will be the angle of the rotation, while ``phi`` the angle that the rotation axis forms with x axis.
```

Comment:
In principle, we could leave the `Args:` structure in the docstring here, as it was before.

Reply:
Yes, we could.

This is something I'm doing on my own: frequently the `Args:` (and equivalents) contain redundant information (essentially repeating the variable names), and redundant type hints.

It was not the case of this docstring in particular, as it is essentially unchanged. But I'm trying to keep the style more discursive.
However, I'm not pushing to enforce it. Not now, at least.